### PR TITLE
Dev/android/comment

### PR DESCRIPTION
**Pull request by musasimsekdotdll**
See issues #486 and #518.

### CODE REVIEW

File: android/lib/pages/profile_page.dart [hunk 1]
```diff
@@ -588,7 +601,15 @@ class _ProfilePageState extends State<ProfilePage> {
                                                   user_account_info
                                                               .profile_picture_id ==
                                                           null
-                                                      ? profilePictureBuilder(3)
+                                                      ? CircleAvatar(
```
Comment (canatakan) [inline android/lib/pages/profile_page.dart:604]: Do not use those. Instead use `circleAvatarBuilder()` in `get_image_builder.dart` as we discussed. There are a lot of problems with image processing in this code.

File: android/lib/pages/profile_page.dart [hunk 2]
```diff
@@ -719,12 +740,15 @@ Widget profilePictureBuilder(picture_id) {
                 return const Text(
                     "An error occured while loading profile page!");
               }
-
+              String final_string = image_output.image!.base64String;
```
Comment (canatakan) [inline android/lib/pages/profile_page.dart:743]: Parsing the images like these are prone to errors, as mentioned in the above comment, use `circleAvatarBuilder()`.

File: android/lib/pages/settings_page.dart [hunk 3]
```diff
@@ -60,7 +60,7 @@ class _SettingsPageState extends State<SettingsPage> {
   }
 
   Scaffold settingsPage(GetSettingsOutput settings) {
-    Widget profilePicture = circleAvatarBuilder(settings.profilePictureId!, 10);
+    Widget profilePicture = circleAvatarBuilder(settings.profilePictureId, 10);
```
Comment (canatakan) [inline android/lib/pages/settings_page.dart:63]: Removing this null check is a correct approach. Missed this during my implementation.

File: android/lib/widgets/comment.dart [hunk 4]
```diff
@@ -78,110 +96,113 @@ class CommentListWidget extends StatefulWidget {
...
+                Row(
+                  children: [
+                    comments.value[index].authorAccountInfo.profile_picture_id == null
+                        ? CircleAvatar(
+                      radius: 12.0,
+                      backgroundColor: Colors.grey[300],
+                      backgroundImage:
+                      MemoryImage(base64Decode(defaultbase64)),
+                    )
+                        : profilePictureWidget(
```
Comment (canatakan) [inline android/lib/widgets/comment.dart:134]: This widget is apparently problematic. Consider updating this as well.

**Review by canatakan — CHANGES_REQUESTED** [review]
The functionality is broken (most likely because of the processing of image data). Please check my comments and consider using `circleAvatarBuilder()` in `get_image_builder.dart` as we discussed.

File: android/lib/pages/profile_page.dart (hunk 2 above)
Comment (musasimsekdotdll) [reply android/lib/pages/profile_page.dart]: This widget is not used anymore but it returns a circleAvatarBuilder() just in case

**Review by canatakan — APPROVED** [review]
* Thank you for updating your code according to changes. The image issues are resolved.
* Profile view bugs are fixed.
* Art Item overflow is fixed.
* Verification bar is hidden in the profile for the other users.

I approve your changes and I am merging your updates.

**Pull request merged**